### PR TITLE
use env var to switch next.js to turbopack mode

### PR DESCRIPTION
**Pull request by sokra**
### What?

this forces all tests to use turbopack independent of the way they invoke next dev

### Why?

some tests were not running turbopack

fixes WEB-1187

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 31f33cb66f58c9ad3ca7198d118ac6d0d1648881

`pnpm testheadless test/development/basic/styled-components-disabled.test.ts` 
- styled-components SWC transform > should have hydration mismatch with styled-components transform disabled

<details>
<summary>Expand output</summary>

  ● styled-components SWC transform › should have hydration mismatch with styled-components transform disabled

    TIMED OUT: success

    [object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]

    undefined

      569 |
      570 |   if (hardError) {
    > 571 |     throw new Error('TIMED OUT: ' + regex + '\n\n' + content + '\n\n' + lastErr)
          |           ^
      572 |   }
      573 |   return false
      574 | }

      at check (lib/next-test-utils.js:571:11)
      at Object.<anonymous> (development/basic/styled-components-disabled.test.ts:50:7)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: test/integration/custom-routes/test/index.test.js [hunk 1]
```diff
@@ -2732,38 +2732,6 @@ describe('Custom routes', () => {
     runTests(true)
   })
 
-  // enable once https://github.com/vercel/turbo/pull/3894 is landed
-  describe.skip('dev mode (turbo)', () => {
```
Comment (sokra) [inline test/integration/custom-routes/test/index.test.js:2736]: We don't need that. Instead we can add custom-routes to the tests-manifest to enable that

File: test/lib/next-test-utils.js [hunk 2]
```diff
@@ -921,38 +919,16 @@ export function shouldRunTurboDevTest() {
     return false
   }
 
-  const shouldRunTurboDev = !!process.env.__INTERNAL_NEXT_DEV_TEST_TURBO_DEV
+  const shouldRunTurboDev = !!process.env.TURBOPACK
   // short-circuit to run all the test with --turbo enabled skips glob matching costs
   if (shouldRunTurboDev) {
     console.log(
-      `Running tests with --turbo via custom environment variable __INTERNAL_NEXT_DEV_TEST_TURBO_DEV`
+      `Running tests with turbopack via custom environment variable TURBOPACK`
```
Comment (alexkirsz) [inline test/lib/next-test-utils.js:926]:
```suggestion
      `Running tests with Turbopack via custom environment variable TURBOPACK`
```

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -118,7 +118,7 @@ jobs:
     uses: ./.github/workflows/build_reusable.yml
     with:
       skipForDocsOnly: 'yes'
-      afterBuild: RUST_BACKTRACE=0 NEXT_EXTERNAL_TESTS_FILTERS="$(pwd)/packages/next-swc/crates/next-dev-tests/tests-manifest.js" __INTERNAL_NEXT_DEV_TEST_TURBO_DEV=TRUE __INTERNAL_CUSTOM_TURBOPACK_BINDINGS="$(pwd)/packages/next-swc/native/next-swc.linux-x64-gnu.node" __INTERNAL_NEXT_DEV_TEST_TURBO_GLOB_MATCH="*" NEXT_E2E_TEST_TIMEOUT=240000 NEXT_TEST_MODE=dev node run-tests.js --type development --timings -c ${TEST_CONCURRENCY}
+      afterBuild: RUST_BACKTRACE=0 NEXT_EXTERNAL_TESTS_FILTERS="$(pwd)/packages/next-swc/crates/next-dev-tests/tests-manifest.js" TURBOPACK=1 __INTERNAL_CUSTOM_TURBOPACK_BINDINGS="$(pwd)/packages/next-swc/native/next-swc.linux-x64-gnu.node" NEXT_E2E_TEST_TIMEOUT=240000 NEXT_TEST_MODE=dev node run-tests.js --type development --timings -c ${TEST_CONCURRENCY}
```
Comment (alexkirsz) [inline .github/workflows/build_and_test.yml:121]: Nit: Why not `NEXT_TURBOPACK` instead? This seems like a Next.js specific option.

**Review by alexkirsz — APPROVED** [review]

**Pull request merged**